website/upgrade-guides/0-14.html.markdown

```diff
@@ -321,12 +321,32 @@ instead of the actual value.
 Terraform v0.14 introduces a more extensive version of that behavior where
 Terraform will track when you write an expression whose result is derived
 from a
-[sensitive input variable](/docs/configuration/outputs.html#sensitive-suppressing-values-in-cli-output),
+[sensitive input variable](/docs/configuration/outputs.html#sensitive-suppressing-values-in-cli-output) or
 [sensitive output value](/docs/configuration/variables.html#suppressing-values-in-cli-output),
-or an attribute defined as sensitive by a provider,
 and so after upgrading to Terraform v0.14 you may find that more values are
 obscured in the Terraform plan output than would have been in Terraform v0.13.
 
+If a sensitive value (either derived from a sensitive input variable or a sensitive output variable) is used in another module output, that output must be marked `sensitive` as well to be explicit about this data being passed through Terraform:
+
+```terraform
+variable "foo" {
+  sensitive = true
+}
+
+output "bar" {
+  value     = var.foo
+  sensitive = true # sensitive must be true when referencing a sensitive input variable
+}
+```
+
+There is also experimental behavior that will extend this sensitivity-awareness to attributes providers define as sensitive. You can enable this feature by activating the experiment in the `terraform` block:
+
+```
+terraform {
+  experiments = [provider_sensitive_attrs]
+}
+```
```

Comment:
I worry a bit that this experiment might not be easy to understand without an example. Is it feasible to add some sample config here which can be used to explain what will change when the experiment is toggled?

Maybe something like this:

```hcl
data "vault_generic_secret" "foobar" {
  path = "secret/foobar"
}

resource "random_pet" "pet" {
  prefix = data.vault_generic_secret.foobar.data["prefix"]
}

output "token" {
  value = data.vault_generic_secret.foobar.data["token"]
}
```

> With the experiment enabled, the `vault_generic_secret` resource's `data` attribute is marked sensitive throughout the configuration. This means that the plan diff will suppress the value of the `random_pet` resource's `prefix` attribute, and the output must have its `sensitive` attribute set to `true`.

Reply:
I agree, an example would make this easier to understand. 

Checking to be sure because I honestly don't use data sources much ... your example data source here `vault_generic_secret` here doesn't have `sensitive` defined, https://github.com/terraform-providers/terraform-provider-vault/blob/00de4240f1c6eda3fb0cbfab9ea15f6ca1381fa9/vault/data_source_generic_secret.go#L38 am I right? I am not calling out, I honestly am not sure, because when I check the _resource_ `vault_generic_secret`'s source code, I can find an attr that is defined as sensitive, so I want to ask.

Reply:
The `data` and `data_json` attributes have been marked sensitive for a couple of months now: https://github.com/terraform-providers/terraform-provider-vault/blob/f500c71c7000fb7b00bfc54dee05eb74fc4f9454/vault/data_source_generic_secret.go#L39-L44

The change was [made in this PR](https://github.com/terraform-providers/terraform-provider-vault/pull/844) and released in v2.13.0.

Reply:
Thanks! I got sucked into some weird changeset while browsing GitHub UI, it seems. Thanks!